File: OOPS_LAB/Program_6/student.java
class StudentDetails{
    private String name;
    private int roll;
    private int[] marks = new int[3];

    public StudentDetails(String nameI,int rollI, int[] scores)
    {
        name=nameI;
        roll=rollI;
        for(int i=0; i<scores.length; i++)
        {
            this.marks[i]=scores[i];
        }
    }

    private  int totalMarks()
    {
        int total=0;
        for(int i=0; i<marks.length; i++)
        {
            total+=marks[i];
        }
        return total;
    }

    private  double avgMarks()
    {
        double avg =(double)totalMarks()/marks.length;
        return avg;
    }

    public void display()
    {
        System.out.println("Name : "+name);
        System.out.println("Roll No : "+roll);
        System.out.println("Subject\tMarks");
        for(int i=0; i<marks.length; i++)
        {
            System.out.println("SUB"+(i+1)+"\t"+marks[i]);
        }
        System.out.println("Total Marks "+ totalMarks());
        System.out.println("Avg Marks "+ avgMarks());
    }
}

class student{
    public static void main(String[] args) {
        int[] marks = {30,40,50};
        StudentDetails std1 = new StudentDetails("Ashfak", 3004, marks);
        std1.display();
    }
}

File: OOPS_LAB/Program_6/student.cpp
#include<iostream>
#include<vector>
using namespace std;

class Student {
private:
    string name;
    int roll;
    vector<int> marks;

    int totalMarks() {
        int total = 0;
        for (auto x : marks) {
            total += x;
        }
        return total;
    }

    float avgMarks() {
        if (marks.empty()) return 0;
        return (float)totalMarks() / marks.size();
    }

public:
    Student(string name, int roll, vector<int> m) : name(name), roll(roll), marks(m) {}

    void display() {
        cout << "Name : " << name << endl;
        cout << "Roll No : " << roll << endl;
        cout << "Subject\tMarks" << endl;

        for (int i = 0; i < marks.size(); i++) {
            cout << "SUB" << (i + 1) << "\t" << marks[i] << endl;
        }

        cout << "Total Marks: " << totalMarks() << endl;
        cout << "Avg Marks: " << avgMarks() << endl;
    }
};

int main() {
    vector<int> marks = {50, 60, 70};
    
    Student std1("Ashfak", 253004, marks);
    std1.display();

    return 0;
}